metadata-load/src/main/java/uk/ac/ebi/ampt2d/metadata/importer/ObjectsImporter.java

```diff
@@ -201,17 +214,21 @@ public ReferenceSequence importReferenceSequence(String accession) {
         Taxonomy taxonomy;
         try {
             // Reference sequences must always be imported through API, even with import mode = DB
-            String referenceSequenceXml = sraxmlRetrieverByAccessionForceApi.getXml(accession);
-            if (referenceSequenceXml.contains(ASSEMBLY_END_TAG)) {
-                AssemblyType assembly = sraAssemblyXmlParser.parseXml(referenceSequenceXml, accession);
-                referenceSequence = referenceSequenceConverter.convert(assembly);
-                taxonomy = taxonomyRepository.findOrSave(extractTaxonomyFromAssembly(assembly));
-            } else if (referenceSequenceXml.contains(ENTRY_END_TAG)) {
-                referenceSequence = sraEntryXmlParser.parseXml(referenceSequenceXml, accession);
-                taxonomy = taxonomyRepository.findOrSave(referenceSequence.getTaxonomy());
+            final String gcfAccessionPattern = "GCF_[\\d]*\\.?[\\d]*";
+            if (accession.matches(gcfAccessionPattern)) {
+                referenceSequence = retrieveReferenceSequenceFromEntrez(accession);
```

Comment:
This branch should follow the same structure as the SRA based one, first using a Entrez retriever (which could store the API key as an attribute), then the parser.